packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -694,6 +749,21 @@ export function prepareToReadContext(
   }
 }
 
+export function readContextAndCompare<C>(
+  context: ReactContext<C>,
+  compare: (C => mixed) | null,
+): C {
+  if (!enableLazyContextPropagation) {
+    return readContext(context);
+  }
```

Comment:
I get why this makes sense logically but seems more appropriate maybe to just make this an error since it's not really valid to build the new flag without the old flag. seems almost certain that this would cause CI to fail if one were to test a build with that particular flag configuration

Reply:
Updated to error if both enableLazyContextPropagation and enableContextProfiling don't pass